lbaas/origin_server_pool_test.go

```diff
@@ -0,0 +1,188 @@
+package lbaas
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-oracle-terraform/helper"
+	"github.com/hashicorp/go-oracle-terraform/opc"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test the Origin Server Pool lifecycle to create, get, delete a Origin Server
+// Pool and validate the fields are set as expected.
+func TestAccOriginServerPoolLifeCycle(t *testing.T) {
+	helper.Test(t, helper.TestCase{})
+
+	// CREATE Parent Load Balancer Service Instance
+
+	var region string
+	if region = os.Getenv("OPC_TEST_LBAAS_REGION"); region == "" {
+		region = "uscom-central-1"
+	}
+	lb := createParentLoadBalancer(t, region, "acc-test-lb-server-pool1")
+
+	// CREATE Origin Server Pool
+
+	serverPoolClient, err := getOriginServerPoolClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	healthCheckInfo := HealthCheckInfo{
+		Type:                "HTTP",
+		Path:                "/health",
+		AcceptedReturnCodes: []string{"2xx", "3xx", "4xx", "5xx"},
+		Enabled:             "TRUE",
+		Interval:            30,
+		Timeout:             30,
+		HealthyThreshold:    6,
+		UnhealthyThreshold:  3,
+	}
+
+	createOriginServerPoolInput := &CreateOriginServerPoolInput{
+		Name: "acc-test-server-pool1",
+		OriginServers: []CreateOriginServerInput{
+			CreateOriginServerInput{
+				Status:   "ENABLED",
+				Hostname: "example.com",
+				Port:     3691,
+			},
+		},
+		HealthCheck: &healthCheckInfo,
+		Tags:        []string{"tag3", "tag2", "tag1"},
+		Status:      "ENABLED",
+	}
+
+	_, err = serverPoolClient.CreateOriginServerPool(lb, createOriginServerPoolInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer destroyOriginServerPool(t, serverPoolClient, lb, createOriginServerPoolInput.Name)
+
+	// FETCH
+
+	resp, err := serverPoolClient.GetOriginServerPool(lb, createOriginServerPoolInput.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := &OriginServerPoolInfo{
+		Name: createOriginServerPoolInput.Name,
+		OriginServers: []OriginServerInfo{
+			OriginServerInfo{
+				Status:   "ENABLED",
+				Hostname: "example.com",
+				Port:     3691,
+			},
+		},
+		HealthCheck: healthCheckInfo,
+		Status:      createOriginServerPoolInput.Status,
+		Tags:        createOriginServerPoolInput.Tags,
+	}
+
+	// compare resp to expected
+	assert.Equal(t, expected.Name, resp.Name, "Origin Server Pool name should match")
```

Comment:
can you just do a deep struct comparison ro do you need to individually check all the fields?

`assert.Equal(expected, resp)`

Reply:
can't compare the whole response as Equal as the ordering of the `Tags` attribute can vary, so needs the `assert.ElementsMatch()` assertion instead of `assert.Equal()` - however the `OriginServers` and `HealthCheck `child structures can be compared rather that each attribute. Optimized other test assertions as well.